Adds multiple pages to the Onboarding chapter (#302)
BitcoinDesign/Guide: guide/onboarding/protecting-a-wallet.md

## Making the toggle less obvious

Having the show/hide button right on the main screen makes things quite obvious for someone who has access to your device to press unhide. A solution would be to move the hide toggle away from the home screen and into the [app settings](https://medium.com/@olanrewajusodiq64/the-ui-ux-of-hide-balance-designing-to-improve-asset-security-e4b20668f315). This way, if someone has access to your device and opens the app, they may not immediately know how to reveal the balance, transaction and/or addresses as it is not made obvious as the previous solutions.

{% include picture.html
   image = "/assets/images/guide/onboarding/protecting-a-wallet/hide-in-settings.gif"
   alt-text = "A gif showing a mobile interface where the toggle for hiding information is moved to settings"
   width = 1600
   height = 800
%}

An advantage here is the risk of an unauthorized person revealing their information is minimal due to the fact that the toggle isn't immediately visible on the home screen. The downside is that a user cannot quickly hide their information if the need arises.

## Why are these patterns important to designers?

> Privacy in bitcoin payments goes far beyond hiding balances and other sensitive information, the [privacy by design framework](https://www.ipc.on.ca/wp-content/uploads/Resources/7foundationalprinciples.pdf) states that privacy should be incorporated and built into products by default. This way, whether or not the user is concerned with their data privacy, they would always be protected through good UX and UI.

Reviewer: ```suggestion
Privacy in bitcoin payments goes far beyond hiding balances and other sensitive information. The [privacy by design framework](https://www.ipc.on.ca/wp-content/uploads/Resources/7foundationalprinciples.pdf) states that privacy should be incorporated and built into products by default. This way, whether or not the user is concerned with their data privacy, they would always be protected through good UX and UI.
```

-->

{% include picture.html
   image = "/assets/images/guide/onboarding/protecting-a-wallet/hide-sensitive-information-header.png"
   retina = "/assets/images/guide/onboarding/protecting-a-wallet/hide-sensitive-information-header.png"
   mobile ="/assets/images/guide/onboarding/protecting-a-wallet/hide-sensitive-information-header.png"
   mobileRetina ="/assets/images/guide/onboarding/protecting-a-wallet/hide-sensitive-information-header.png"
   alt-text = "Payment Privacy & Security"
   caption = 'Person by [DrawKit](https://www.drawkit.io/illustrations/mobile-article-colour){:target="_blank" rel="nofollow"} and trees by [Flaticon](https://www.flaticon.com/free-icon/tree-with-many-leaves_25267){:target="_blank" rel="nofollow"}'
   width = 1600
   height = 600
   layout = "full-width"
%}

# Protecting a wallet

Bitcoin wallets contain sensitive information such as recovery phrases, balances, transaction histories, and more. Therefore, it is crucial to add extra layers of security to keep private data out of bad actors' hands. Additional protection may also be necessary for users who share devices, such as parents with children.

> A common practice for wallets is to have a barrier (i.e. Touch ID, PINs, etc.) that must be unlocked to open a wallet application. Creating robust and challenging- to- crack passwords and PIN codes isn't something that everyone excels at. In order to decrease the chance of someone’s wallet getting hacked because of poor security decisions, you can implement the following into your design:

Reviewer 2: ```suggestion
A common practice for wallets is to have a barrier (i.e., Touch ID, PINs, etc.) that must be unlocked to open a wallet application. Creating robust and challenging-to-crack passwords and PIN codes isn't something that everyone excels at. To decrease the chance of someone’s wallet getting hacked because of poor security decisions, you can implement the following into your design:
```

### Touch ID/Face ID

<div class="center" markdown="1">

{% include image.html
   image = "/assets/images/guide/onboarding/protecting-a-wallet/face-id-info.png"
   retina = "/assets/images/guide/onboarding/protecting-a-wallet/face-id-info@2x.png"
   width = 250
   height = 250
   alt-text = "Face ID screen"
   caption = "Face ID UI from [Wallet UI Kit](https://www.figma.com/file/VB3GQdAnhl8yta44DY3PSV/Bitcoin-Wallet-UI-Kit?node-id=1228%3A27860)"
   layout = "float-left-desktop -background -shadow"
%}

One convenient way for users to verify their identity to unlock your wallet app is with biometrics. This could take the form of either touch or face ID.

> Users of Android and iOS devices are super familiar with these motions. More importantly, using biometrics does not require the user to remember anything, furthermore decreasing their chances of getting locked out of their wallet.

Reviewer 2: ```suggestion
Users of Android and iOS devices are already familiar with these motions. More importantly, using biometrics does not require the user to remember anything, decreasing their chances of getting locked out of their wallet.
```

Unfortunately, biometrics are not supported on all mobile devices. Hardcore security users may be wary about giving up their biometric information.

</div>

{% include /tip/open.html label="Tip: Be Transparent" icon="info" color="blue" %}

Bitcoin wallet users may feel hesitant about the privacy implications of your product storing touch/face ID information. Be sure your app thoroughly relays how this information is stored.

{% include /tip/close.html %}

### PIN

> A PIN is one of the most common and traditional methods for securing a mobile application. This setup requires that users create a 4-6 digit PIN that must be entered when opening their wallet application, or when viewing sensitive information such as their recovery phrase.

Reviewer 2: ```suggestion
A PIN is one of the most common and traditional methods for securing a mobile application. This setup requires that users create a 4-6 digit PIN that must be entered when opening their wallet application or viewing sensitive information such as their recovery phrase.
```

<div class="image-slide-gallery">

{% include picture.html
   image = "/assets/images/guide/onboarding/protecting-a-wallet/choose-pin.png"
   retina = "/assets/images/guide/onboarding/protecting-a-wallet/choose-pin@2x.png"
   alt-text = "Choosing a pin screen"
   width = 250
   height = 541
   class = "shadow"
%}

{% include picture.html
   image = "/assets/images/guide/onboarding/protecting-a-wallet/enter-pin.png"
   retina = "/assets/images/guide/onboarding/protecting-a-wallet/enter-pin@2x.png"
   alt-text = "Enter pin screen"
   width = 250
   height = 541
   class = "shadow"
%}

</div>

> The PIN-based approaches can be supported by all devices, and therefore is more accessible. A PIN can also be used to encrypt data within a wallet, which makes any locally stored information on the device inaccessible. Additionally, some users prefer this type of security measure because they use the same or similiar PINs on multiple applications/devices, making it easy to remember.

Reviewer 2: ```suggestion
All devices can support the PIN-based approach, which makes it more accessible. A PIN can also be used to encrypt data within a wallet, making any locally stored information on the device inaccessible. Additionally, some users prefer this type of security measure because they use the same or similar PINs on multiple applications/devices, making it easy to remember.
```

> However, users with poor security practices might create easy-to-guess PINs (such as 1234), which would decrease the level of security. This design also relies on users to remember their PIN correctly.

Reviewer 2: ```suggestion
However, users with poor security practices might create easy-to-guess PINs (such as 1234), which would provide a false sense of security. This design also relies on users to remember their PIN correctly.
```

{% include tip/open.html color="green" icon="check" label="Do: Educate on Good Security Practices" %}

> Within your product upon PIN creation, tell your users what a good, secure PIN is (i.e. not their birthday). Recommend against (or even make it impossible) to create a PIN such as 1234, 1111, etc.

Reviewer 2: ```suggestion
Within your product upon PIN creation, tell your users what a good, secure PIN is (i.e., not their birthday). Recommend against (or even make it impossible) to create a PIN such as 1234, 1111, etc.
```

{% include tip/close.html %}

### Hiding Sensitive Information

> Imagine this scenario. You are in a public place, and you need to make a payment using your bitcoin wallet. You open your wallet on your phone, but you don’t feel comfortable having your address and balance information clearly visible to strangers who may be looking over your shoulder, persons lurking or video surveillance . Hence by giving users the ability to hide sensitive information in their wallet if desired, they gain an added sense of physical privacy and security [^3]when using the app in public.

Reviewer 2: ```suggestion
Imagine this scenario. You are in a public place, and you need to make a payment using your bitcoin wallet. You open your wallet on your phone, but you don’t feel comfortable having your address and balance information clearly visible to strangers who may be looking over your shoulder, persons lurking, or video surveillance. Hence by giving users the ability to hide sensitive information in their wallet if desired, they gain an added sense of physical privacy and security [^3]when using the app in public.
```

> ### What information is considered sensitive ?

Reviewer 2: We already talk about what information is sensitive, private in the intro paragraph, perhaps merge in the information?

> Bitcoin wallets contain sensitive information such as recovery phrases, balances, transaction histories, and more. Therefore, it is crucial to add extra layers of security to keep private data out of bad actors' hands. Additional protection may also be necessary for users who share devices, such as parents with children.

Not a blocker, but could be improved to avoid repetition.

> Sensitive information in wallet applications include the wallet balance, addresses, private keys and previous transactions information.

Reviewer 2: ```suggestion
Sensitive information in wallet applications includes the wallet balance, addresses, private keys, and previous transactions information.
```

- Wallet Balance - shows how much you own
- Addresses - can be used to track your transaction history
- Private keys - can be used to access and transfer your bitcoins

> It's more common for wallets to protect private keys, but not much is done for other sensitive information like the balance, addresses, and previous transactions. A few wallets like Bitcoin Core, Wasabi, Wallet of Satoshi, and others have made it work, though. Below is a pattern and considerations for hiding and revealing sensitive information.

Reviewer 3: Link out to those referenced wallets?

> It's more common for wallets to protect private keys but not much is done for other sensitive information like the balance, addresses and previous transactions. A few wallets like Bitcoin Core, Wasabi, Wallet of Satoshi, and others have made it work though. Below describes a pattern, and considerations for hiding and revealing sensitive information.

Reviewer 2: ```suggestion
It's more common for wallets to protect private keys, but not much is done for other sensitive information like the balance, addresses, and previous transactions. A few wallets like Bitcoin Core, Wasabi, Wallet of Satoshi, and others have made it work, though. Below is a pattern and considerations for hiding and revealing sensitive information.
```

### Quickly hide from overview screen

> The hide icon / button, which is usually displayed within close reach of the balance itself, is used to quickly and easily hide and reveal wallet information by tapping or clicking on it.[^2]

Reviewer 2: ```suggestion
The hide icon/button, which is usually displayed within close reach of the balance itself, is used to quickly and easily hide and reveal wallet information by tapping or clicking on it.[^2]
```

{% include picture.html
   image = "/assets/images/guide/onboarding/protecting-a-wallet/hide-by-eye-icon.gif"
   retina = "/assets/images/guide/onboarding/protecting-a-wallet/hide-by-eye-icon.gif"
   mobile = "/assets/images/guide/onboarding/protecting-a-wallet/hide-by-eye-icon.gif"
   mobileRetina = "/assets/images/guide/onboarding/protecting-a-wallet/hide-by-eye-icon.gif"
   alt-text = "hide-by-eye icon"

>    width = 1600

Reviewer 3: Image size are wrong on these GIFs.

Reviewer 3: No need for the retina, mobile and mobileRetina attributes if they all specify the same image.

Reviewer 3: Also needs proper alt text, not just the file name.

Author: 🚧

   height = 800
%}

> #### Pros

Reviewer 3: I'd just use plain text instead of Pros and Cons on this page. Each technique just has one pro or con, and they are each just one sentence. No need for lots of bullet lists that contain just one item.

- It is quite convenient to switch between revealed and hidden state

#### Cons

- Easy for anyone else to reveal your information if they have access to your device.

> ## Entering a PIN to reveal information

Reviewer 3: The header hierarchy is off here. This header should probably not be H2 but something lower.

> ## Entering a PIN to reveal information

Reviewer 2: Does this slightly feel like a duplication of the "Pin" section?

Author: Yeah could argue that. I think it's okay though as this provides a more specific case for revealing a users balance. Initially is more for just locking the whole app.

Reviewer 2: Agreed, not really a blocker for the current phase for sure.

> A problem that seemed to arise throughout each implementation was that it was as easy to reverse the hidden state as it was to enable it. This may be for convenience sake, however, for protection against unauthorized access, perhaps you should only be able to unhide your information if a PIN or password has been entered. This could therefore reaffirm the identity of the wallet owner for extra security.

Reviewer 3: > A problem that seemed to arise throughout each implementation...

What is meant by that? Who was implementing what? Maybe some outdated text that should be removed?

The second sentence is also weird with the reference to `you`.

{% include picture.html
   image = "/assets/images/guide/onboarding/protecting-a-wallet/reveal-by-pin.gif"
   retina = "/assets/images/guide/onboarding/protecting-a-wallet/reveal-by-pin.gif"
   mobile = "/assets/images/guide/onboarding/protecting-a-wallet/reveal-by-pin.gif"
   mobileRetina = "/assets/images/guide/onboarding/protecting-a-wallet/reveal-by-pin.gif"
   alt-text = "reveal-by-PIN"
   width = 1600
   height = 800
%}

#### Pros

> - The risk of an unauthorized person revealing your information is minimal due to the PIN required

Reviewer 3: ```suggestion
- The risk of an unauthorized person revealing a users information is minimal due to the PIN requirement
```

#### Cons

> - It may not be convenient for the users to repeatedly put in their PIN when ever they want to reveal your information especially if they do so often.

Reviewer 3: ```suggestion
- It may not be convenient for users to repeatedly enter their PIN to reveal sensitive information, especially if they do this often.
```

###  Hide if inactive

> Another solution would be to invoke the wallet’s hidden state as a default when the app is opened. This means that if someone sees your wallet when you open it, they wouldn't be able to make out the information. The pre-hidden state can be unveiled after a tap, PIN entry, or perhaps a short 5-second timer.

Reviewer 3: ```suggestion
Another solution is to invoke the wallet’s hidden state as a default when the app is opened, to protect against prying eyes during initial display. The pre-hidden state can be unveiled after a tap, PIN entry, or perhaps a short 5-second timer.
```

> Another solution would be to invoke the wallet’s hidden state as a default when the app is opened. What this means is that in the event that someone sees your wallet when you open it, they wouldn't be able to make out the information. The pre-hidden state can be unveiled  after a tap, PIN entry, or perhaps a short 5 second timer.

Reviewer 2: ```suggestion
Another solution would be to invoke the wallet’s hidden state as a default when the app is opened. This means that if someone sees your wallet when you open it, they wouldn't be able to make out the information. The pre-hidden state can be unveiled after a tap, PIN entry, or perhaps a short 5-second timer.
```

{% include picture.html
   image = "/assets/images/guide/onboarding/protecting-a-wallet/hide-by-default.gif"
   retina = "/assets/images/guide/onboarding/protecting-a-wallet/hide-by-default.gif"
   mobile = "/assets/images/guide/onboarding/protecting-a-wallet/hide-by-default.gif"
   mobileRetina = "/assets/images/guide/onboarding/protecting-a-wallet/hide-by-default.gif"
   alt-text = "hide-by-default"
   width = 1600
   height = 800
%}

#### Pros

> - Users have some time to assess their environment before their info is displayed.

Reviewer 3: ```suggestion
- Users have time to assess their environment before their info is displayed.
```

#### Cons

> - The user may feel a bit frustrated having to wait for their information to be revealed especially in an urgent situation

Reviewer 3: ```suggestion
- Users may get frustrated having to wait for their information to be revealed, especially in urgent situations
```

### Making the toggle less obvious

> Having the show/hide button right on the main screen makes things quite obvious for someone who has access to your device to press unhide. A solution would be to move the hide toggle away from the home screen and into the app settings[^1]. This way, if someone has access to your device and opens the app, they may not immediately know how to reveal the balance, transaction and/or addresses as it is not made obvious as the previous solutions.

Reviewer 3: ```suggestion
Having the show/hide button right on the main screen makes it quite obvious for someone who has access to a users device to reveal the information. A solution would be to move the button away from the home screen and into the app settings[^1]. This way, if someone has access to a users device and opens the app, they may not immediately know how to reveal the balance, transaction and/or addresses, as it is not made obvious as in the previous solutions.
```

> Having the show/hide button right on the main screen make things quite obvious for someone who has access to your device to press unhide. A solution would be to move the hide toggle away from the home screen and into the app settings[^1]. This way, if someone has access to your device and opens the app, they may not immediately know how to reveal the balance, transaction and/or addresses as it is not made obvious as the previous solutions.

Reviewer 2: ```suggestion
Having the show/hide button right on the main screen makes things quite obvious for someone who has access to your device to press unhide. A solution would be to move the hide toggle away from the home screen and into the app settings[^1]. This way, if someone has access to your device and opens the app, they may not immediately know how to reveal the balance, transaction and/or addresses as it is not made obvious as the previous solutions.
```

{% include picture.html
   image = "/assets/images/guide/onboarding/protecting-a-wallet/hide-in-settings.gif"
   retina = "/assets/images/guide/onboarding/protecting-a-wallet/hide-in-settings.gif"
   mobile = "/assets/images/guide/onboarding/protecting-a-wallet/hide-in-settings.gif"
   mobileRetina = "/assets/images/guide/onboarding/protecting-a-wallet/hide-in-settings.gif"
   alt-text = "hide-in-settings"
   width = 1600
   height = 800
%}

#### Pros

> - The risk of an unauthorized person revealing your information is minimal due to the fact that the toggle isn't immediately visible on the home screen

Reviewer 3: ```suggestion
- The risk of an unauthorized person revealing a users information is minimal due to the fact that the toggle isn't immediately visible on the home screen
```

#### Cons

> - This solution may not be convenient for a user to quickly hide their information if the need arises.

Reviewer 3: ```suggestion
- This solution may not be convenient for users to quickly hide their information if the need arises
```

### Why is this pattern important to designers?

Privacy in bitcoin payments goes far beyond hiding balances and other sensitive information, the privacy by design framework[^4] states that privacy should be incorporated and built into products by default. This way, whether or not the user is concerned with their data privacy, they would always be protected through good UX and UI.

> By including the hiding information pattern in design of wallets, we give users a greater sense of control and comfortability in any environment when using it. They have the freedom to decide whether or not they want their information visible.

Reviewer 3: ```suggestion
By including UX patterns for hiding information pattern wallets, we give users a greater sense of control and comfort in any situation. They have the freedom to decide whether or not they want their information visible.
```

Now let's look at the different ways a user might go about [funding a wallet]({{ 'guide/onboarding/funding-a-wallet' | relative_url }}).

> [^1]: https://medium.com/@olanrewajusodiq64/the-ui-ux-of-hide-balance-designing-to-improve-asset-security-e4b20668f315

Reviewer 3: There's no need for footnotes on this page. Should be just links.

   width = 250
   height = 541
%}

{% include picture.html
   image = "/assets/images/guide/onboarding/protecting-a-wallet/enter-pin.png"
   retina = "/assets/images/guide/onboarding/protecting-a-wallet/enter-pin@2x.png"
   alt-text = "Enter pin screen"
   width = 250
   height = 541
%}

</div>

Pros: PIN-based approaches can be supported by all devices, and therefore is more accessible. A PIN can also be used to encrypt data within a wallet, which makes any locally stored information on the device inaccessible. Additionally, some users prefer this type of security measure because they use the same or similiar PINs on multiple applications/devices, making it easy to remember.

Cons: Users with poor security practices might create easy-to-guess PINs (such as 1234), which would decrease the level of security. This design also relies on users to remember their PIN correctly.

{% include tip/open.html color="green" icon="check" label="Do: Educate on Good Security Practices" %}

Within your product upon PIN creation, tell your users what a good, secure PIN is (i.e. not their birthday). Recommend against (or even make it impossible) to create a PIN such as 1234, 1111, etc.

> 

Reviewer 3: Add link to the next page.

> {% include tip/close.html %}

Reviewer 4: Something you could add here that I think is useful is having the option to create a 'hidden' wallet for plausible deniability. This is something probably best setup internally as an extra security measure but probably worth mentioning. BlueWallet is an example of a mobile wallet with this feature

Author: Great idea, thinking about implementing the work Tiana did [here](https://github.com/BitcoinDesign/Guide/pull/112)

Author: Added as part of the "protecting a wallet" page.
